components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java

```diff
@@ -444,30 +447,34 @@ private boolean validateTopicName(String topicName) {
     @Override
     public String[] getBackendRoles() throws EventBrokerException {
         UserRealm userRealm = CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+        String[] cleanedRoles = new String[0];
```

Comment:
In a case of there's only admin role, is this method supposed to return new String[0] ?